src/request.ts

```diff
@@ -23,14 +25,21 @@ export type ExcludeObject<T> = T extends object ? (object extends T ? never : T)
  * @typedef Request
  * @param {MethodDescriptor} methodDescriptor
  * @param {RequestParams} requestParams, defaults to an empty object ({})
```

Comment:
Noob-question: 
Seems like  `RequestParams` could be extended with a `context` field, to support this kind of feature without adding a whole new object to thread through the code. 

Any reasons for not doing that instead? Safety? Is it a bit unclear how consumers have extended the `requestParams` before 🤔

Reply:
Ex. this could work like `host` or `timeout` does currently:
```
request.enhance({ context: { message: 'hello from mw1' }}))
...
const { message } = request.context()
```

Reply:
Thanks for the suggestion!

But adding stuff to `RequestParams` is tricky since it will be used when synthesizing the outgoing request. I do not want to add a "special key" there that has to be accounted for across the code base, that will be difficult to maintain and bug prone. 

Examples: 
- `Request#params` is using requestParams in its entirety (which includes `host` and `timeout`) - this is used in `Request#path` and `Request#pathTemplate` and eventually in `Request#url` which is called by the HTTP gateway: `const defaults = url.parse(this.request.url())`. So it might mean that the context ends up in the requested url.
- The code for `Request#enhance` is also already really obscure if the above was not bad enough.